vertx-web/src/main/java/io/vertx/ext/web/handler/sse/impl/SSEConnectionImpl.java

```diff
@@ -0,0 +1,182 @@
+package io.vertx.ext.web.handler.sse.impl;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.VertxException;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.eventbus.MessageConsumer;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.sse.SSEConnection;
+import io.vertx.ext.web.handler.sse.SSEHeaders;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+public class SSEConnectionImpl implements SSEConnection {
+
+	private static final String MSG_SEPARATOR = "\n";
+	private static final String PACKET_SEPARATOR = "\n\n";
+
+	private final RoutingContext context;
+	private boolean rejected;
+	private List<MessageConsumer> consumers = new ArrayList<>();
+
+	public SSEConnectionImpl(RoutingContext context) {
+		this.context = context;
+	}
+
+	@Override
+	public SSEConnection forward(String address) {
+		consumers.add(context.vertx().eventBus().consumer(address, this::ebMsgHandler));
+		return this;
+	}
+
+	@Override
+	public SSEConnection forward(List<String> addresses) {
+		consumers = addresses.stream().map(address ->
```

Comment:
You might lose track of consumers previously registered

Reply:
So should we remove the part dealing with the event bus, simply?

Reply:
I meant that if you add new addresses to be forwarded, you will lose track of consumers previously registered because the consumers value is overwritten. Then when the `SSEConnection` is closed some consumers could be leaked.

Reply:
I don't understand this point.
There's a list of eventbus consumers attached to this connection. if you forward multiple eventbus addresses to this connection, we'll unregister them all when the connection gets closed.
Could you be more specific please. I'm sure I'm missing the point.
The best would be to write a unit test reproducing the "leak" you're mentioning thanks.